cogs/member_stats.py

```diff
@@ -1,21 +1,82 @@
 import discord
-from discord.ext import commands
+from discord.ext import tasks, commands
 from pathlib import Path
 from .validation import is_moderator, is_mod_commands_channel
 import json
+import os
+from os import path
+from datetime import datetime
+from dateutil import tz
 
 
 class MemberStats(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
+        self.status = 0
+
+    def cog_unload(self):
+        self.m_stats.cancel()
+
+    @commands.command()
+    async def m_stats_start(self, ctx):
+        self.m_stats.start(ctx)
+        self.status = 1
+        await ctx.send(f'statistics recording started')
+
+    @commands.command()
+    async def m_stats_stop(self, ctx):
+        self.m_stats.cancel()
+        self.status = 0
+        await ctx.send(f'statistics recording stopped')
+
+    @commands.command()
+    async def m_stats_status(self, ctx):
+        if self.status == 1:
+            await ctx.send(f'statistics recording is running')
+        else:
+            await ctx.send(f'statistics recording NOT running')
 
     @commands.command()
+    async def m_stats_read(self, ctx):
+        if self.status == 1:
+            await ctx.send(f'first stop statistics recording ($m_stats_stop)')
+        else:
+            if path.exists('./server_specific/member_stats.csv'):
+                with open('./server_specific/member_stats.csv',
+                          'r') as stats_file:
+                    contents = stats_file.read()
+                    await ctx.send('note: recorded timezone is PST (UTC -8)')
+                    await ctx.send(f'{contents}')
+                    stats_file.close()
+            else:
+                await ctx.send(f'file wiped')
+
+    @commands.command()
+    async def m_stats_clear(self, ctx):
+        def check(m):
+            return (
+                not m.content.startswith('$m')
+            )
+        await ctx.send(f'wipe ALL statistics? y(yes) / n(no)')
+
```

Comment:
As the user is interacting with `Seng` here, then I believe they should be added to `processes`. I haven't documented this well, if at all, so please feel free to message me on Discord about it!

Otherwise, here's a brief summary:
Effectively, if Seng is going to interact with a user within a single command (i.e. a single command triggers any conversation between Seng and the user), then Seng needs to do the following:

 - Check that the user is not already in a conversation with Seng (check `processes`)
    - If they are, then Seng cannot start a new conversation with them.
    - If they are not, then Seng continues, but adds that user to `processes`.
 - During the conversation, Seng should not count any other `$m` messages from the user, which you've already covered. Effectively, this means that the user can issue non-conversation commands to Seng to reach extra information while they're in a conversation, without that conversation being interrupted.
 - At the end of the conversation, Seng must then remove that user from `processes`, so that the user can have the ability to go do other 'conversation' commands.
 
 I implemented this some times in the `send_welcome.py`. You could read an example here and it might help a bit: https://github.com/NootNootEliot/seng/blob/75ffdc26eb564a72e24ddd915ddec40a9cef020a/cogs/send_welcome.py#L397